src/utils/sortAndFilterResults.ts

```diff
@@ -0,0 +1,111 @@
+import { SearchParametersType, UnprocessedResultsFromCRM, OwnerType } from '../types'
+
+type MatchTallies = {
+  [index: string]: number
+  neighbour: number
+  propertyType: number
+  propertyGroup: number
+}
+
+function matchForPropertyTypes (property: UnprocessedResultsFromCRM, desiredPropertyTypes: string[]): {propertyTypeUpdateTally: number, propertyTypeMatch: boolean} {
+    let propertyTypeUpdateTally = 0
+    let propertyTypeMatch = false
+    desiredPropertyTypes.forEach((propertyType: string) => {
```

Comment:
Array.prototype.some could be useful here. I don't think you need to return a number here.

```
return desiredPropertyTypes.some((propertyType: string) => {
        return propertyTypeMatch || desiredPropertyTypes.includes('All') || property.Property_Category_Mailing.includes(propertyType);
})
```

You're always going to increase by one so can do
```
if (propertyTypeMatch) matchTallies.propertyType += 1
```